### PR TITLE
Use Microsoft.Playwright to set up Docker test container

### PR DESCRIPTION
**Pull request by louis-z**
Plus, remove dependency on AngleSharp

### CODE REVIEW

File: NGitLab.Tests/Docker/GitLabDockerContainer.cs [hunk 1]
```diff
@@ -278,127 +275,94 @@ private async Task GenerateCredentialsAsync()
...
-                    result = await SignIn(context);
+                    await page.Locator("form#new_user input[name='user[login]']").FillAsync(AdminUserName);
+                    await page.Locator("form#new_user input[name='user[password]']").FillAsync(AdminPassword);
+
+                    var checkbox = page.Locator("form#new_user input[type=checkbox][name='user[remember_me]']");
+                    await checkbox.CheckAsync(new LocatorCheckOptions { Force = true });
+
+                    await page.RunAndWaitForResponseAsync(async () =>
+                    {
+                        await page.EvalOnSelectorAsync("form#new_user", "form => form.submit()");
```
Comment (Toa741) [inline NGitLab.Tests/Docker/GitLabDockerContainer.cs:308]: Could it simply be replaced with a click on the form submit button?

File: NGitLab.Tests/Docker/GitLabDockerContainer.cs [hunk 2]
```diff
@@ -278,127 +275,94 @@ private async Task GenerateCredentialsAsync()
...
+                    }, GitLabUrl.AbsoluteUri);
+
+                    url = await GetCurrentUrl(page);
```
Comment (Toa741) [inline NGitLab.Tests/Docker/GitLabDockerContainer.cs:311]: page.Url doesn't work here?

File: NGitLab.Tests/Docker/GitLabDockerContainer.cs (hunk 1 above)
Comment (meziantou) [reply NGitLab.Tests/Docker/GitLabDockerContainer.cs:308]: Clicking on a button means more maintenance. If the UI change => KO.

File: NGitLab.Tests/Docker/GitLabDockerContainer.cs (hunk 2 above)
Comment (louis-z) [reply NGitLab.Tests/Docker/GitLabDockerContainer.cs:311]: It partially works... but sometimes with a delay. For instance, say you make these 3 calls in a row

```cs
url = page.Url;
url = await GetCurrentUrl(page);
url = page.Url;
```

- the 1st time around, all 3 calls will return the same thing: `/users/sign_in`
- the 2nd time around, the 1st call will return `/users/sign_in`, but the 2nd and 3rd calls will return `/`

**Pull request merged**